src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -100,6 +108,1124 @@
+					// generate an empty migration record with error and move on
+					String errorMessage = Utils.NO_CTOOLS_SITE + " " + siteId/*site name*/;
+					handleBulkResourceBoxRootFolderError(userId, bulkMigrationName,
+							bulkMigrationId, siteId, siteName, "default_resource_tool_id",
+							Utils.TOOL_NAME_RESOURCES, errorMessage);
+					continue;
+				}
+				
+				try
+				{
+					// add admin user to site with Owner role
+					// will remove the user from site later once migration is done.
+					addAdminAsSiteOwner(ctoolsAdminUserName, siteId, sessionId);
+					
+					String toolId = "";
+					String toolName = "";
+	
+					// 2. get tool id, tool name
+					HashMap<String, String> pagesMap = get_user_project_site_tools(siteId, sessionId);
+					String pagesString = pagesMap.get("pagesString");
+					JSONArray pagesJSON = new JSONArray(pagesString);
+					for (int pageIndex = 0; pageIndex < pagesJSON.length(); pageIndex++) {
+						JSONObject pageJSON = (JSONObject) pagesJSON.get(pageIndex);
+						// look the tools attribute and find resource tool
+						JSONArray toolsJSON = (JSONArray) pageJSON.get("tools");
+						for (int toolIndex = 0; toolIndex < toolsJSON.length(); toolIndex++) {
+							JSONObject toolJSON = (JSONObject) toolsJSON
+									.get(toolIndex);
+							if (migrationToolId.equals(toolJSON.get("toolId"))) {
+								// found Resources tool
+								toolId = (String) toolJSON.get("id");
+								toolName = (String) toolJSON.get("title");
+							}
+						}
+					}
+					
+					if (migrationToolId.equals(Utils.MIGRATION_TOOL_RESOURCE))
+					{
+						// check to see whether the site has any resource
+						if (checkSiteWithEmptyResourceList(sessionId, userId, bulkMigrationName,
+								bulkMigrationId, siteId,
+								siteName, toolId, toolName))
+						{
+							// stop here if site has no resource
+							// remove added admin user
+							removeAddedAdminOwner(siteId);
+							continue;
+						}
+						
+						// bulk migration of resource items into Box
+						siteBoxMigrationIdMap = createRootBoxFolderWithMembers(sessionId,
+								userId, bulkMigrationName,
+								bulkMigrationId, siteId, siteName, toolId, toolName, siteBoxMigrationIdMap);
+					}
+					else if (migrationToolId.equals(Utils.MIGRATION_TOOL_EMAILARCHIVE))
+					{
+						// DO NOTHING FOR NOW, since we are not doing email migration
+						// bulk migration of email archive messages into Google Groups
+						/*handleBulkMessageGoogleMigration(sessionId,
+								request, response, userId, bulkMigrationName,
+								bulkMigrationId, siteId, siteName, toolId, toolName);
+						*/
+					}
+					else
+					{
+						// wrong tool
+						log.error(" unrecognized migration tool " + migrationToolId);
+					}
+				}
+				catch (Exception e)
+				{
+					log.info("uploadBatch exception " + e.getMessage() + " for site " + siteId);
+					
+					// remove added admin user
+					removeAddedAdminOwner(siteId);
```

Comment:
The admin user is needed for the following content transfer. It will be removed in the very end when the migration is finished for this site.